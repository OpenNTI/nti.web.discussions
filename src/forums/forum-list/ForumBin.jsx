import React from 'react';
import PropTypes from 'prop-types';

import ForumBoard from './ForumBoard';

export default function ForumBin ({ title, bin }) {
	return (
		<li className="bin">
			{title && <h2 className="bin-title">{title}</h2>}
			<ul className="forum-board-list">
				{
					Object.keys(bin).map((key, i, boards) => (
						<ForumBoard
							title={!title && key === 'Section' && boards.length === 1 ? '' : key}
							board={bin[key]}
							key={key}
						/>
					))
				}
			</ul>
		</li>
	);
}

ForumBin.propTypes = {
	/**
	*	A bin is an object with boards. Currently the usual bins
	* 	are "Enrolled For-Credit", "Open Discussions", "Other Discussions"
	*	{
	*		Parent: {
	*			forums: [...]
	*		},
	*		Section: {
	*			forums: [...]
	*		}
	*	}
	*/
	bin: PropTypes.object.isRequired,

	/**
	* The localized title for this bin (e.g. "Enrolled For-Credit")
	*/
	title: PropTypes.string.isRequired,
};
