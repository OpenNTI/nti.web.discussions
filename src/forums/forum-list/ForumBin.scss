@import '~@nti/style-common';

.discussion-forum-list {
	.bin {
		h2.bin-title {
			margin: 0;
			padding: 0.5rem 0.9375rem;
			font: normal 400 1.5rem/1.3 var(--legacy-header-font-family);
			color: var(--secondary-grey);
		}

		> ul.forum-board-list {
			list-style: none;
			margin: 0;
			padding: 0;
		}
	}
}
