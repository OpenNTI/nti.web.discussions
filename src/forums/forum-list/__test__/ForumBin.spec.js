/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';

import ForumBin from '../ForumBin';

describe('Forum Bin', () => {
	test('Snapshot', () => {
		const bin = {
			Section: {
				forums: [
					{
						MimeType:
							'application/vnd.nextthought.forums.communityforum',
						NTIID:
							'tag:nextthought.com,2011-10:unknown-OID-0x0551b0:5573657273:ZkMpTMuJZhf',
						TopicCount: 0,
						title: 'One',
						getID: () =>
							'tag:nextthought.com,2011-10:unknown-OID-0x0551b0:5573657273:ZkMpTMuJZhf',
						edit: () => {},
						hasLink: () => true,
						Links: [],
					},
					{
						MimeType:
							'application/vnd.nextthought.forums.communityforum',
						NTIID:
							'tag:nextthought.com,2011-10:unknown-OID-0x0551b8:5573657273:ZkMpTMuJZhe',
						TopicCount: 0,
						title: 'Two',
						getID: () =>
							'tag:nextthought.com,2011-10:unknown-OID-0x0551b8:5573657273:ZkMpTMuJZhe',
						edit: () => {},
						hasLink: () => true,
						Links: [],
					},
				],
			},
			Parent: {
				forums: [
					{
						NTIID:
							'tag:nextthought.com,2011-10:unknown-OID-0x05513f:5573657273:ZkMpTMuJZj8',
						MimeType:
							'application/vnd.nextthought.forums.communityforum',
						TopicCount: 2,
						title: 'Parent One',
						edit: () => {},
						hasLink: () => true,
						getID: () =>
							'tag:nextthought.com,2011-10:unknown-OID-0x05513f:5573657273:ZkMpTMuJZj8',
						Links: [],
					},
					{
						NTIID:
							'tag:nextthought.com,2011-10:unknown-OID-0x055199:5573657273:ZkMpTMuJZhn',
						MimeType:
							'application/vnd.nextthought.forums.communityforum',
						TopicCount: 0,
						title: 'Parent Two',
						edit: () => {},
						hasLink: () => true,
						getID: () =>
							'tag:nextthought.com,2011-10:unknown-OID-0x055199:5573657273:ZkMpTMuJZhn',
						Links: [],
					},
				],
			},
		};
		const tree = renderer.create(<ForumBin title="" bin={bin} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
