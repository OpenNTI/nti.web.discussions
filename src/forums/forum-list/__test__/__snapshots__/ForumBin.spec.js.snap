// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forum Bin Snapshot 1`] = `
<li
  className="bin"
>
  
  <ul
    className="forum-board-list"
  >
    <li
      className="board"
    >
      <h3
        className="board-title"
      >
        My Section
      </h3>
      <ul
        className="forum-item-list"
      >
        <li
          className="forum-item-li"
        >
          <div
            className="pathLink nti-link-to-path forum-link"
            to="/id/unknown-OID-0x0551b0%3A5573657273%3AZkMpTMuJZhf"
          >
            <div
              className="item-container"
            >
              <div
                className="item-main"
              >
                <span
                  className="title"
                />
                <div
                  className="meta"
                >
                  <span
                    className="see-all count"
                  >
                    No Discussions
                  </span>
                </div>
              </div>
            </div>
          </div>
        </li>
        <li
          className="forum-item-li"
        >
          <div
            className="pathLink nti-link-to-path forum-link"
            to="/id/unknown-OID-0x0551b8%3A5573657273%3AZkMpTMuJZhe"
          >
            <div
              className="item-container"
            >
              <div
                className="item-main"
              >
                <span
                  className="title"
                />
                <div
                  className="meta"
                >
                  <span
                    className="see-all count"
                  >
                    No Discussions
                  </span>
                </div>
              </div>
            </div>
          </div>
        </li>
      </ul>
    </li>
    <li
      className="board"
    >
      <h3
        className="board-title"
      >
        All Sections
      </h3>
      <ul
        className="forum-item-list"
      >
        <li
          className="forum-item-li"
        >
          <div
            className="pathLink nti-link-to-path forum-link"
            to="/id/unknown-OID-0x05513f%3A5573657273%3AZkMpTMuJZj8"
          >
            <div
              className="item-container"
            >
              <div
                className="item-main"
              >
                <span
                  className="title"
                />
                <div
                  className="meta"
                >
                  <span
                    className="see-all count"
                  >
                    2 Discussions
                  </span>
                </div>
              </div>
            </div>
          </div>
        </li>
        <li
          className="forum-item-li"
        >
          <div
            className="pathLink nti-link-to-path forum-link"
            to="/id/unknown-OID-0x055199%3A5573657273%3AZkMpTMuJZhn"
          >
            <div
              className="item-container"
            >
              <div
                className="item-main"
              >
                <span
                  className="title"
                />
                <div
                  className="meta"
                >
                  <span
                    className="see-all count"
                  >
                    No Discussions
                  </span>
                </div>
              </div>
            </div>
          </div>
        </li>
      </ul>
    </li>
  </ul>
</li>
`;
